File: cloud-functions/demo-node-api/test-timeout-14.js
// 测试 maxDuration: sleep 15s，超过 edgeone.json 中配置的 10s 限制
export default async function onRequest() {
  const start = Date.now();
  await new Promise((resolve) => setTimeout(resolve, 14000));
  return new Response(JSON.stringify({ elapsed: Date.now() - start }), {
    headers: { 'Content-Type': 'application/json' },
  });
}

File: cloud-functions/demo-sanic/test-timeout-14.py
"""测试 maxDuration: sleep 14s，超过 edgeone.json 中配置的 python.maxDuration=8s 限制"""
import time
import json


def handler(request):
    start = time.time()
    time.sleep(14)
    elapsed = time.time() - start
    return {
        "statusCode": 200,
        "headers": {"Content-Type": "application/json"},
        "body": json.dumps({"elapsed": round(elapsed, 2), "unit": "seconds"})
    }
